packages/metrics/src/features/locationStatistics/service.ts

```diff
@@ -67,10 +67,10 @@ async function cacheOfficeCount(authHeader: IAuthHeader) {
   ])
   locations.forEach(({ id }) => (OFFICE_COUNT_CACHE[id] = -1))
   const locationsMap = locations.reduce<LocationsMap>(
-    (locationsMap, location) => ({
-      ...locationsMap,
-      [location.id]: location
-    }),
+    (locationsMap, location) => {
+      locationsMap[location.id] = location
+      return locationsMap
+    },
```

Comment:
:ok_hand:

Reply:
Can we write this as a good old for loop and leave a comment why it's necessary to do this way? I'm sure next time me or someone else lands here they have the urge to rewrite this in an immutable way 😄

Reply:
I managed to find a way that's better than a for loop but also immutable :smile:

Reply:
Nice one